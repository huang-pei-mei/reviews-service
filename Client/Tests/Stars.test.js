import 'jsdom-global/register';
import React from 'react';
import { shallow, mount, render } from 'enzyme';
import "@babel/polyfill";
import Stars from '../src/Components/stars.js';
import mockSuccessAjax from './mockSuccessAjax';
import successData  from './successData.json';
import renderer from 'react-test-renderer';

it ('should render the stars component correctly', () => {
  let review = mockSuccessAjax(successData).slice(0, 1);
  const tree = renderer.create(<Stars review={review}/>).toJSON();

  expect(tree).toMatchSnapshot();
})

it ('shouldRender the number of overallStars', () => {
  let review = mockSuccessAjax(successData).slice(0, 1);
  const starBody = shallow(<Stars review={review} />);

  expect(starBody.find('.overallStars').exists()).toBe(true);
})

it ('should not render if double data is passed in', () => {
  let review = {overalstars: 5, storyStars: 4, performanceStars: 5};
  const starBody = shallow(<Stars review={review} />);

  expect(starBody.find('.overallStars')).toMatchSnapshot();
})