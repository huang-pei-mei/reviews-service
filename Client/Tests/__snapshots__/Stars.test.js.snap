// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render if double data is passed in 1`] = `ShallowWrapper {}`;

exports[`should render the stars component correctly 1`] = `
<div
  className="reviewAndStars"
>
  <div>
    <span
      className="overallStarsText"
    >
      Overall               
    </span>
    <span
      className="Stars overallStars"
    />
  </div>
  <div>
    <span
      className="performanceStarsText"
    >
      Performance       
    </span>
    <span
      className="Stars performanceStars"
    />
  </div>
  <div>
    <span
      className="storyStarsText"
    >
      Story                   
    </span>
    <span
      className="Stars storyStars"
    />
  </div>
</div>
`;
